src/Controller/OmniaController.php

```diff
@@ -141,282 +268,218 @@ public function videoList() {
    * @throws \Exception
    */
   public function video(Request $request) {
-
-    $response = new JsonResponse();
-    $content = $request->getContent();
-    $query = $this->mediaEntityStorage()->getQuery();
-
-    $token = $request->query->get('token', NULL);
-    if (NULL !== $token) {
-
-      $config = $this->config('nexx_integration.settings');
-      if ($token != $config->get('notification_access_key')) {
-        throw new AccessDeniedHttpException();
-      }
-    }
-    else {
+    if (!$this->isTokenValid($request)) {
+      $this->logger->error('Wrong or missing token in request.');
       throw new AccessDeniedHttpException();
     }
 
-    if (!empty($content)) {
-      $videoData = json_decode($content);
-    }
-    if (!isset($videoData->itemID)) {
+    $content = $request->getContent();
+    $this->videoData = json_decode($content);
+
+    if (!$this->getVideoId()) {
       throw new \Exception('ItemID missing');
     }
 
     $this->logger->info('Incoming video "@title" (nexx id: @id)',
       [
-        '@title' => $videoData->itemData->title,
-        '@id' => $videoData->itemID,
+        '@title' => (string) $this->videoData->itemData->general->title,
```

Comment:
Check for not empty / isset of this property

Reply:
Property is mandatory, so must be there.